src/main/java/org/generationcp/middleware/util/Util.java

```diff
@@ -490,4 +494,29 @@ public static boolean isLeapYear(final int year) {
 		return isLeapYear;
 	}
 
+
+	public static boolean isDateMatchPattern(final String value, final String dateFormat) {
+		final String strPattern;
+		if (Util.DATE_AS_NUMBER_FORMAT_KSU.equals(dateFormat)) {// d/M/yy|yyyy
+			strPattern = "^(0?[1-9]|[12][0-9]|3[01])\\/(0?[1-9]|1[012])\\/((19|20)\\d\\d|\\d\\d)";
+		} else {// yyyyyMMdd
+			strPattern = "^((19|20)\\d\\d)(0?[1-9]|1[012])(0?[1-9]|[12][0-9]|3[01])";
+		}
+		final Pattern pattern = Pattern.compile(strPattern);
```

Comment:
If this method is called in a loop, Pattern.compile would be costly. Move this to class variable so that pattern is only compiled once.